ceno_zkvm/src/scheme/mock_prover.rs

```diff
@@ -543,94 +543,67 @@ impl<'a, E: ExtensionField + Hash> MockProver<E> {
-                    };
-
-                    lkm
-                });
-
-            let lkm_from_cs = lkm_from_cs.into_finalize_result();
-            let lkm_from_assignment = lkm_from_assignment.into_finalize_result();
-
-            // Compare each LK Multiplicity.
-
-            for (rom_type, cs_map, ass_map) in
-                izip!(ROMType::iter(), &lkm_from_cs, &lkm_from_assignment)
-            {
-                if *cs_map != *ass_map {
-                    let cs_keys: HashSet<_> = cs_map.keys().collect();
-                    let ass_keys: HashSet<_> = ass_map.keys().collect();
-
-                    // lookup missing in lkm Constraint System.
-                    ass_keys.difference(&cs_keys).for_each(|k| {
-                        let count_ass = ass_map.get(k).unwrap();
-                        errors.push(MockProverError::LkMultiplicityError {
-                            rom_type,
-                            key: **k,
-                            count: *count_ass as isize,
-                            inst_id: 0,
-                        })
-                    });
-
-                    // lookup missing in lkm Assignments.
-                    cs_keys.difference(&ass_keys).for_each(|k| {
-                        let count_cs = cs_map.get(k).unwrap();
-                        errors.push(MockProverError::LkMultiplicityError {
-                            rom_type,
-                            key: **k,
-                            count: -(*count_cs as isize),
-                            inst_id: 0,
+            let lkm_from_cs = {
+                let result = transpose(
+                    cb.cs
+                        .lk_expressions_items_map
+                        .iter()
+                        .map(|(rom_type, items)| {
+                            (
+                                *rom_type,
+                                items
+                                    .iter()
+                                    .map(|expr| {
+                                        let insts =
+                                            wit_infer_by_expr(&[], wits_in, pi, &challenge, expr)
+                                                .get_base_field_vec()
+                                                .iter()
+                                                .map(|f| f.to_canonical_u64())
+                                                .collect::<Vec<u64>>();
+                                        (expr.clone(), insts)
+                                    })
+                                    .collect::<Vec<(Expression<E>, Vec<u64>)>>(),
+                            )
                         })
-                    });
+                        .collect::<Vec<(ROMType, Vec<(Expression<E>, Vec<u64>)>)>>(),
+                );
+                // Remove the last instance, which is the padding instance.
+                let instances = result.len();
```

Comment:
The logic to retrieve padding instance is somehow incorrect. 

The correct way to retrieve number of padding row is via https://github.com/scroll-tech/ceno/blob/cec7b82c29f0d06041bf20aec3e3c0932e6643fa/ceno_zkvm/src/scheme/mock_prover.rs#L789

and https://github.com/scroll-tech/ceno/blob/cec7b82c29f0d06041bf20aec3e3c0932e6643fa/ceno_zkvm/src/scheme/mock_prover.rs#L829